src/bindgen.rs

```diff
@@ -3,26 +3,53 @@
 use emoji;
 use error::Error;
 use progressbar::Step;
-use std::path::Path;
+use std::path::{Path, PathBuf};
 use std::process::Command;
+use which::which;
 use PBAR;
 
 /// Install the `wasm-bindgen` CLI with `cargo install`.
-pub fn cargo_install_wasm_bindgen(step: &Step) -> Result<(), Error> {
+pub fn cargo_install_wasm_bindgen(
+    path: &Path,
+    version: &str,
+    install_permitted: bool,
+    step: &Step,
+) -> Result<(), Error> {
+    // If the `wasm-bindgen` dependency is already met, print a message and return.
+    if wasm_bindgen_path(path)
+        .map(|bindgen_path| wasm_bindgen_version_check(&bindgen_path, version))
+        .unwrap_or(false)
+    {
+        let msg = format!("{}WASM-bindgen already installed...", emoji::DOWN_ARROW);
+        PBAR.step(step, &msg);
+        return Ok(());
+    }
+
+    // If the `wasm-bindgen` dependency was not met, and installs are not
+    // permitted, return a configuration error.
+    if !install_permitted {
+        let msg = format!("WASM-bindgen v{} is not installed!", version);
+        return Error::crate_config(&msg);
+    }
+
     let msg = format!("{}Installing WASM-bindgen...", emoji::DOWN_ARROW);
     PBAR.step(step, &msg);
     let output = Command::new("cargo")
         .arg("install")
-        .arg("wasm-bindgen-cli")
         .arg("--force")
+        .arg("wasm-bindgen-cli")
+        .arg("--version")
+        .arg(version)
```

Comment:
My understanding is that this will keep them pinned to the same version as the package, but if they use a new one in a different project it'll overwrite it?

How do we account for versions such as "0.2" which allows for any version from ">= 0.2 && < 0.3"? which version should be chosen in that case? The one in the lockfile? I only realized we'd have to answer this question just now unfortunately :/

Reply:
Oh I see now further below you're specifying to install it locally and to use that one like was discussed. Awesome :D I still wonder about the versioning in the second part of my comment.

Reply:
@mgattozzi Regarding the second part of your first comment, would this crate be a sensible solution? https://crates.io/crates/semver

Still thinking about how its `VersionReq` struct would be used to find the most recent version that satisfies a requirement like `>= 0.2 && < 0.3`, but it seems like a promising library for this.

Reply:
That is the canonical semver crate. I'm just thinking like:

I build a program using "0.2" as my version
Cargo.lock has it at 0.2.8

A new release is out and wasm-bindgen is at 0.2.9

Which one do you use? The lockfile or new crates.io? I think because it's a library the newest one on crates.io that fits the description, but that also means running cargo update which may not be desired behavior. I'm more asking this question to you and also @ashleygwilliams and whoever else might have a stake in answering this.

Reply:
I think we should always follow the lock file. Since we do (or will do) a `cargo build` before getting the `wasm-bindgen` CLI, the lock file will always have the exact wasm-bindgen used to build the `.wasm` binary.

Reply:
Following the lock file makes a lot of sense to me! If it makes sense to everybody else, I could try adjusting the `get_wasm_bindgen_version` function so that it looks through the `Cargo.lock` file rather than the `.toml` file.

If I *do* go this route, should `get_wasm_bindgen_version` end up in a new file, i.e. `lockfile.rs` given that we wouldn't be checking the manifest itself?